Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/BaseClientSubmitter.cs

```diff
@@ -161,13 +167,21 @@ public Output GetTaskOutputInfo(string taskId)
   ///   to start until all dependencies are completed successfully
   /// </summary>
   /// <param name="payloadsWithDependencies">A list of Tuple(taskId, Payload) in dependence of those created tasks</param>
+  /// <param name="maxRetries">The number of retry before fail to submit task</param>
   /// <returns>return a list of taskIds of the created tasks </returns>
   [UsedImplicitly]
-  public IEnumerable<string> SubmitTasksWithDependencies(IEnumerable<Tuple<byte[], IList<string>>> payloadsWithDependencies)
-    => SubmitTasksWithDependencies(payloadsWithDependencies.Select(payload => Tuple.Create(Guid.NewGuid()
-                                                                                               .ToString(),
-                                                                                           payload.Item1,
-                                                                                           payload.Item2)));
+  public IEnumerable<string> SubmitTasksWithDependencies(IEnumerable<Tuple<byte[], IList<string>>> payloadsWithDependencies,
+                                                         int                                       maxRetries = 5)
+    => payloadsWithDependencies.ToChunk(ChunkSize)
+                               .SelectMany(chunk =>
+                                           {
+                                             return ChunkSubmitTasksWithDependencies(chunk.Select(payload => Tuple.Create(Guid.NewGuid()
```

Comment:
`payload` is not an understable name.
 Prefer `payloadWithDependencies` or something similar